apps/web/src/components/Settings/Profile/Profile.tsx

```diff
@@ -215,76 +215,151 @@ const ProfileSettingsForm: FC<ProfileSettingsFormProps> = ({ profile }) => {
     } catch {}
   };
 
+  const uploadAndSave = async () => {
+    if (!currentProfile) {
+      return toast.error(Errors.SignWallet);
+    }
+    const croppedImage = await getCroppedImg(imageSrc, croppedAreaPixels);
+    if (!croppedImage) {
+      return toast.error(Errors.SomethingWentWrong);
+    }
+
+    try {
+      setUploading(true);
+      const ipfsUrl = await uploadCroppedImage(croppedImage);
+      const dataUrl = croppedImage.toDataURL('image/png');
+      setCoverIpfsUrl(ipfsUrl);
+      setUploadedImageUrl(dataUrl);
+    } catch (error) {
+      toast.error(t`Upload failed`);
+    } finally {
+      setShowCropModal(false);
+      setUploading(false);
+    }
+  };
+
+  const onFileChange = async (evt: ChangeEvent<HTMLInputElement>) => {
+    const file = evt.target.files?.[0];
+    if (file) {
+      setImageSrc(await readFile(file));
+      setShowCropModal(true);
+    }
+  };
+
   const isLoading =
-    isUploading || typedDataLoading || dispatcherLoading || signLoading || writeLoading || broadcastLoading;
+    isUploading ||
+    typedDataLoading ||
+    dispatcherLoading ||
+    signLoading ||
+    writeLoading ||
+    broadcastLoading ||
+    uploading;
+
+  const coverPictureUrl = profile?.coverPicture?.original?.url;
+  const coverPictureIpfsUrl = coverPictureUrl ? imageProxy(sanitizeDStorageUrl(coverPictureUrl), COVER) : '';
 
   return (
-    <Card className="p-5">
-      <Form
-        form={form}
-        className="space-y-4"
-        onSubmit={({ name, location, website, twitter, bio }) => {
-          editProfile(name, location, website, twitter, bio);
-        }}
+    <>
+      <Modal
```

Comment:
Add modal to last 🙇  instead of start in the component

Reply:
done!